DEVELOPMENT.md

```diff
@@ -0,0 +1,198 @@
+Developing bugsnag-android
+============
+
+The following guide should get you setup for developing bugsnag-android.
+
+# Initial setup
+
+1. Download the [latest stable version](https://developer.android.com/studio) of Android Studio.
+2. Download [r16b of the NDK](https://developer.android.com/ndk/downloads/older_releases)
+3. Set the `$ANDROID_HOME` and `$ANDROID_NDK_HOME` environment variables to point to the SDKs, and optionally add the `adb/android/emulator` [platform tools](https://developer.android.com/studio/command-line/variables) to your `$PATH`
+4. Clone the repository and its submodules: `git submodule update --init --recursive`
+5. Open the project in Android Studio to trigger indexing and downloading of dependencies
+
+# Project structure
+
+The project is comprised of three [Gradle modules](https://docs.gradle.org/current/userguide/multi_project_builds.html):
+
+- [bugsnag-android-core](bugsnag-android-core/README.md), responsible for capturing JVM errors and delivery of payloads
+- [bugsnag-plugin-android-ndk](bugsnag-plugin-android-ndk/README.md), responsible for capturing NDK errors
+- [bugsnag-plugin-android-anr](bugsnag-plugin-android-anr/README.md), responsible for capturing ANRs
+
+[Gradle tasks](https://docs.gradle.org/current/userguide/tutorial_using_tasks.html) can generally be run on the entire project, or on an individual module: 
+
+```shell
+./gradlew build // builds whole project
+./gradlew bugsnag-plugin-android-anr:build // builds one module only
+```
+
+# Static analysis
+
+Several static analysis checks are run against bugsnag-android to maintain the quality of the codebase. `./gradlew lint ktlintCheck detekt checkstyle` runs them all at once.
+
+## Android Lint
+
+[Android Lint](https://developer.android.com/studio/write/lint) Runs Android-specific static analysis checks. Warnings can be suppressed by following [this guide](https://developer.android.com/studio/write/lint#config).
+
+```shell
+./gradlew lint
+```
+
+## Ktlint
+
+[Ktlint](https://github.com/pinterest/ktlint) runs style checks on Kotlin code.
+
+```shell
+./gradlew ktlintCheck
+```
+
+Some violations (but not all) can be automatically addressed by running the following task:
+
+```shell
+./gradlew ktlintFormat
+```
+
+## Detekt
+
+[Detekt](https://github.com/detekt/detekt) runs static analysis checks on Kotlin code.
+
+```shell
+./gradlew detekt
+```
+
+Warnings can be suppressed by running the following task:
+
+```shell
+./gradlew detektBaseline
+```
```

Comment:
Should probably add a warning that this permanently suppresses ALL outstanding detekt warnings currently in the project, including anything the developer hasn't investigated yet! And that this modifies `detekt-baseline.xml`, which must then be committed.

Reply:
Updated to state that this file will be written to.